src/Tribe/Aggregator/Processes/Service_Provider.php

```diff
@@ -16,6 +16,10 @@ public function register() {
 		tribe_register( 'events-aggregator.record-items', 'Tribe__Events__Aggregator__Record__Items' );
 		tribe_register( 'events-aggregator.processes.import-events', 'Tribe__Events__Aggregator__Processes__Import_Events' );
 		tribe_singleton( 'events-aggregator.queue-control', 'Tribe__Events__Aggregator__Processes__Queue_Control' );
+		tribe_singleton(
+			Tribe__Events__Aggregator__Processes__Batch_Imports::class,
+			Tribe__Events__Aggregator__Processes__Batch_Imports::class
```

Comment:
Since this is being bound as singleton, move the line where you're building the `$batch` here and just bind the `$batch` instance (assuming a `use Tribe__Events__Aggregator__Processes__Batch_Imports as Batch_Imports`)

```php
$batch_imports = new Batch_Imports();
tribe_singleton(Batch_Imports::class, $batch_imports);

// ... keep using $batch ... 
```

Reply:
Question: By doing this we are technically not creating a singleton or we can rely on the fact that `register()` is only called once? 

Also what might be the different in using an `instance` instead of the class name as reference to create a singleton instance.

Thanks :)

Reply:
We can rely on service providers `register` method being called only once.
Furthermore, the container will automatically bind an instance as singleton.
The difference is in performance: there's no need to ask the container to resolve to an instance when the instance is already available in the same context. It's not wrong, just a micro-optimization.

Reply:
Great makes sense. Thanks :)